contracts/fees_collector_contract.py

```diff
@@ -188,17 +206,141 @@ def claim_rewards(self, user: Account, proxy: ProxyNetworkProvider):
+
+        return current_week
+    
+    def get_total_energy_for_week(self, proxy: ProxyNetworkProvider, week: int) -> int:
+        data_fetcher = FeeCollectorContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getTotalEnergyForWeek', [week])
+        if not raw_results:
+            return 0
+        return int(raw_results)
+    
+    def get_total_rewards_for_week(self, proxy: ProxyNetworkProvider, week: int):
+        serializer = Serializer("@")
+
+        data_fetcher = FeeCollectorContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getTotalRewardsForWeek', [week])
+        
+        if not raw_results:
+            return {}
+        
+        b = bytes.fromhex(raw_results)
+        
+        data = b
+
+        def create_esdt_token_payment() -> StructValue:
+            return StructValue([
+                Field("token_identifier", StringValue()),
+                Field("token_nonce", U64Value()),
+                Field("amount", BigUIntValue()),
+            ])
+
+        destination = ListValue(
+            item_creator=create_esdt_token_payment
+        )
+
+        s = serializer.deserialize_parts([data], [destination])
+
+        tokens = []
+
+        for i, item in enumerate(destination.items):
+            fields = cast(StructValue, item).fields
+            for field in fields:
+                fvalue: Any = field.value
+                tokens.append([field.name, fvalue.value])
+       
+        tokens_list = [tokens[i:i+3] for i in range(0, len(tokens), 3)] 
+
+        return tokens_list
+    
+    def get_all_stats(self, proxy: ProxyNetworkProvider, user: str, week: int):        
+        fees_collector_stats = {
+            "current_week": self.get_current_week(proxy),
+            "total_rewards_for_week": self.get_total_rewards_for_week(proxy, week),
+            "total_energy_for_week": self.get_total_energy_for_week(proxy, week)
+        }
+
+        return fees_collector_stats
+    
+
+    def get_tx_op(self, tx_hash: str, operation: dict, api: ApiNetworkProvider) -> dict:
+        used_api = api
+        # Get the transaction details
+        tx = used_api.get_transaction(tx_hash)
+        # Get and check transaction operations
+        ops = tx.raw_response['operations']
+        
+
+        tx_decoded = tx
+        new = self.transaction_decoder.get_transaction_metadata(tx_decoded)
+
+        print(new.function_name)
+        print(new.function_args)
+
+        # Take each op in ops and match it with operation. Try to match only the fields expected in operation dictionary. 
+        # TX Operations are unordered. If any of the operations match, return it.
+        for op in ops:
+            # print(f'Matching with {operation}')
+            if all(op.get(key) == operation.get(key) for key in operation.keys()):
+                return op
+            
+    def wrap(self, source: str, width: int):
```

Comment:
What's this used for? I doubt it should be here...

Reply:
Taken out. Was left over from a solution that didn't work